### PR TITLE
10.0 porting website_sale_checkout_skip_payment

### PR DESCRIPTION
**Pull request by eLBati**

### CODE REVIEW

**Review by tafaRU — APPROVED** [review]

File: website_sale_checkout_skip_payment/README.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.svg
+   :target: https://www.gnu.org/licenses/lgpl.html
+   :alt: License: LGPL-3
+
+==================================
+Website Sale Checkout Skip Payment
+==================================
+
+This module allows to logged users to checkout with no payment step. At the
```
Comment (simahawk) [inline website_sale_checkout_skip_payment/README.rst]: "allows logged users"

File: website_sale_checkout_skip_payment/README.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.svg
```
Comment (simahawk) [inline website_sale_checkout_skip_payment/README.rst]: png

File: website_sale_checkout_skip_payment/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+{
+    'name': 'Website Sale Checkout Skip Payment',
+    'summary': 'Skip payment for logged users in checkout process',
+    'version': '10.0.0.1.0',
```
Comment (simahawk) [inline website_sale_checkout_skip_payment/__manifest__.py]: 10.0.1.0.0

File: website_sale_checkout_skip_payment/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    'category': 'Website',
+    'website': 'https://www.tecnativa.com',
```
Comment (simahawk) [inline website_sale_checkout_skip_payment/__manifest__.py]: https://github.com/OCA/e-commerce

File: website_sale_checkout_skip_payment/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            # Clean session, then redirect to the confirmation page
+            request.website.sale_reset()
+            return request.redirect('/shop/confirmation')
+        else:
+            return request.render(
+                'website_sale_skip_payment.confirmation_order_error')
+
+    @http.route()
+    def payment_get_status(self, sale_order_id, **post):
+        # When skip payment step, the transaction not exists so only render
```
Comment (simahawk) [inline website_sale_checkout_skip_payment/controllers/main.py]: transaction does not exists

File: website_sale_checkout_skip_payment/models/website.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+class Website(models.Model):
+    _inherit = "website"
+
+    checkout_skip_payment = fields.Boolean(
+        compute='_compute_checkout_skip_payment')
+
+    @api.multi
+    def _compute_checkout_skip_payment(self):
+        for rec in self:
+            if request.session.uid:
```
Comment (simahawk) [inline website_sale_checkout_skip_payment/models/website.py:18]: just checking: is 100% that this is going to be recomputed for each user?
I'd rather use a std model method to get it not cached.

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by grindtildeath** [comment]
I get this error when confirming an order through web shop :
```
2018-04-27 09:50:45,531 1 ERROR bssfw-24 odoo.http: Exception during JSON request handling.
Traceback (most recent call last):
  File "/opt/odoo/src/odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/src/odoo/http.py", line 677, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/src/odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/src/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/src/odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/src/odoo/http.py", line 935, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/src/odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/external-src/e-commerce/website_sale_checkout_skip_payment/controllers/main.py", line 35, in payment_get_status
    'message': request.website._render(
AttributeError: 'website' object has no attribute '_render'
```

**Comment by rafaelbn** [comment]
Hi @eLBati thanks for this migration! Please could you check comments? Thanks

**Comment by rafaelbn** [comment]
Hello @eLBati could you review comments? Are you interested in continuing this PR?

- The v11 migration is here https://github.com/OCA/e-commerce/pull/228
- We did it duplicated in https://github.com/OCA/e-commerce/pull/246 :yin_yang: 
- You should take care of https://github.com/OCA/e-commerce/pull/223
- Check comments from @simahawk 

Let me know! :smiley:

File: website_sale_checkout_skip_payment/models/website.py (hunk 6 above)
Comment (eLBati) [reply website_sale_checkout_skip_payment/models/website.py:18]: @simahawk just tested with 2 different users on 2 different browsers: it works

**Comment by eLBati** [comment]
@grindtildeath can you describe the steps to reproduce the error? I am not able to reproduce it in a standard scenario

**Comment by eLBati** [comment]
@rafaelbn thanks, https://github.com/OCA/e-commerce/pull/223 is already included with https://github.com/OCA/e-commerce/pull/203/commits/e26b2c5da1c34eaacb5fb88bf59b9f9026f7b8d8

File: website_sale_checkout_skip_payment/README.rst (hunk 1 above)
Comment (eLBati) [reply website_sale_checkout_skip_payment/README.rst]: @simahawk done

File: website_sale_checkout_skip_payment/README.rst (hunk 2 above)
Comment (eLBati) [reply website_sale_checkout_skip_payment/README.rst]: @simahawk readme files re-created with new fragments

File: website_sale_checkout_skip_payment/__manifest__.py (hunk 3 above)
Comment (eLBati) [reply website_sale_checkout_skip_payment/__manifest__.py]: @simahawk done

File: website_sale_checkout_skip_payment/__manifest__.py (hunk 4 above)
Comment (eLBati) [reply website_sale_checkout_skip_payment/__manifest__.py]: @simahawk done

File: website_sale_checkout_skip_payment/controllers/main.py (hunk 5 above)
Comment (eLBati) [reply website_sale_checkout_skip_payment/controllers/main.py]: @simahawk done

**Review by grindtildeath — COMMENTED** [review]
@eLBati Tested again and seems to work, so approved.

**Review by grindtildeath — APPROVED** [review]

**Comment by grindtildeath** [comment]
@eLBati My bad, but while switching off my local instances, I just see I had the same error in my server logs, I'll try to give it another look tomorrow.

**Comment by grindtildeath** [comment]
@eLBati Just tested again and this happens when confirming the order on the webshop with a user having skip payment activated. The UX looks ok, but we get this error in the logs.

**Comment by eLBati** [comment]
@grindtildeath thanks, fixed

**Review by fuentes73 — APPROVED** [review]
Using it in production since a while and everything works OK.

**Comment by zeroheure** [comment]
Why this was never merged ?

**Pull request closed without merging**